pkg/fs/dir.go

```diff
@@ -101,15 +118,57 @@ func readDir(scanDir, namePrefix string) (objInfos []ObjectInfo) {
 		return
 	}
 
-	// Close the directory
+	// Close the directory.
 	f.Close()
 
 	// Sort files by Name.
 	sort.Sort(byName(fis))
 
-	// Populate []ObjectInfo from []FileInfo
+	// Searching for entries with objectName containing marker and prefix.
+	// Binary search is used  for efficient search.
+	var markerIndex, prefixIndex int
+	// Search for the marker entry in the sorted objectInfo array.
+	if queryMarker != "" {
+		// Returns the position where marker could be placed in sorted list.
+		// If there is a matching object name the index of its position is returned.
+		markerIndex = searchFileInfos(fis, queryMarker)
+		// Marker is identified to be at the end of the list.
+		// So returning empty list.
+		if markerIndex == len(fis) {
+			return
+		}
+		// If object name is same the marker, skip that entry.
```

Comment:
Lets say this is the bucket structure. 

## My-bucket 
- obj1
- obj2
- obj3 
- obj4 

IF marker is set to `obj1` , then `obj1` should not appear in the listing.  The result should contain objects starting from `obj2`.